core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -97,8 +97,8 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
             data.map(notificationHandler.processNotification(_)).getOrElse {
```

Comment:
Also, `data` can be `null`, not just `None`. We can maybe take the chance to fix that too.